python/lsst/daf/butler/core/utils.py

```diff
@@ -54,9 +54,10 @@ def iterable(a):
 
     There are three cases, when the input is:
 
-    - iterable, but not a `str` -> iterate over elements
+    - iterable, but not a `str`  or Mapping -> iterate over elements
       (e.g. ``[i for i in a]``)
     - a `str` -> return single element iterable (e.g. ``[a]``)
+    - a Mapping -> return single element iterable
```

Comment:
This makes me a little nervous, as I think it's hard to remember what makes a class qualify as a `Mapping`.  Could we at least make this not the default behavior?  Or maybe allow general `isinstance` exceptions via an optional kwarg, i.e.:
`iterable(x, except=collections.abc.Mapping)` (though that's not terribly readable either).

Reply:
Do you ever want iterable to return the keys of a Mapping? The problem here is dataIds and compatbility with old YAML files.

Reply:
That's a fair point; I think you're right that this is the only behavior we'd ever actually want for mappings - my concern is more that it's not the behavior I would have naively expected.  But it's not a big deal either way.

I do think Python mappings would be much cleaner if they weren't iterable at all and you had to explicitly say `keys()`, but of course there's nothing we can do about that.